subsys/logging/log_backend_xtensa_sim.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2019 Intel Corporation Inc.
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <stddef.h>
+#include <logging/log_backend.h>
+#include <logging/log_core.h>
+#include <logging/log_msg.h>
+#include <logging/log_output.h>
+#include <xtensa/simcall.h>
+
+#define _STDOUT_BUF_SIZE 256
+
+static void preprint_char(int c)
+{
+	char buf[16];
+
+	register int a2 __asm__ ("a2") = SYS_write;
+	register int a3 __asm__ ("a3") = 1;
+	register char *a4 __asm__ ("a4") = buf;
+	register int a5 __asm__ ("a5") = 1;
+	register int ret_val __asm__ ("a2");
+	register int ret_err __asm__ ("a3");
+
+	buf[0] = (char)c;
+	__asm__ volatile ("simcall"
+				: "=a" (ret_val), "=a" (ret_err)
+				: "a" (a2), "a" (a3), "a" (a4), "a" (a5)
+				: "memory");
+}
+
+static u8_t buf[_STDOUT_BUF_SIZE];
+
+static int char_out(u8_t *data, size_t length, void *ctx)
+{
+	for (size_t i = 0; i < length; i++) {
+		preprint_char(data[i]);
```

Comment:
write simcall can output the whole buffer in one invocation

Reply:
thanks, addressed.